modules/tinymce/src/core/main/ts/api/html/SaxParser.ts

```diff
@@ -211,7 +211,7 @@ const checkBogusAttribute = (regExp: RegExp, attrString: string): string | null
  */
 const SaxParser = (settings?: SaxParserSettings, schema = Schema()): SaxParser => {
   settings = settings || {};
-  const doc = settings.document ?? document;
+  const doc = (settings.document ?? document).implementation.createHTMLDocument('parser');
```

Comment:
You probably don't want to do this... you can just use `document` for this use case as it isn't going to matter which version is used. The other thing is this will create a new document every time the content parses... that's not very good for performance. I'd suggest either using the same global or doing something similar as to what's done in the undoManager: https://github.com/tinymce/tinymce/blob/304fecd538e49fc23b5f85284f7dd056887550d9/modules/tinymce/src/core/main/ts/undo/Levels.ts#L12

Reply:
how often do we parse content though 🤔 

I told him to stick with `settings.document` in case it made some sort of difference we needed to retain.

Reply:
Surprisingly often in some cases... either way this is definitely a hot code path so we shouldn't be doing this.

Reply:
@MitchCTiny the change you've made here is no different to the old version, in fact it's a little worse. You're still constructing a new Document every time the SaxParser is initialised. We need to ensure it's only setup once, not every time SaxParser is initialised.

Reply:
Sorry, I misunderstood the method, should be better now